tests/robot-tests/tests/bootstrap_data/release_and_publication_role_permissions_data.robot

```diff
@@ -0,0 +1,64 @@
+#
+# This test suite is responsible for setting up various Publications and Releases for the use of the Publication
+# and Release Permissions UI tests.
+#
+# Releases are created in each approval state and a different role is assigned to Analyst1 for each.  This way,
+# there is a unique Release for each combination of Release approval status available and Release / Publication role.
+#
+
+*** Settings ***
+Resource    bootstrap_common.robot
+Resource    bootstrap_data_constants.robot
+Resource    ../libs/admin-common.robot
+Library     ../libs/admin_api.py
+
+Force Tags  BootstrapData  Local  Dev
+
+Suite Setup       user signs in as bau1
+Suite Teardown    user closes the browser
+
+*** Test Cases ***
+Create test theme and topic
+    Import bootstrap data roles and permissions variables
+    ${THEME_ID}=  user creates theme via api  ${THEME_NAME}
+    ${TOPIC_ID}=  user creates topic via api  ${TOPIC_NAME}  ${THEME_ID}
```

Comment:
Is it necessary to create another theme and topic here instead of use the existing `%{TEST_THEME_NAME}` and `%{TEST_TOPIC_NAME}`?

I think these new themes and topics will quickly accumulate on Dev. The `run_tests.py` script assumes that all the stuff created by the tests is inside the test topic it creates, and it deletes that after the run is complete. And it keeps everything contained within "Test theme", so even if stuff doesn't get cleaned up it all stays contained in there.

The only other test suite that uses the `user creates theme/topic via api` keywords cleans up after itself using `user deletes theme via api` in the suite teardown.

Reply:
It is necessary in this particular case, as this will be a Theme and a Topic that will pre-exist (in seed data and on Dev) and will be readonly.  This is a manually-run (excluded by default) test that actually sets up the data necessary for the UI permissions tests.  The reason being that this is a very time-consuming process that would kill the Robot tests if needing to do it with every test run.

So on Drive, you'll see I've uploaded a new "ees-mssql-data-37.zip" file which contains our original seed data plus the Theme and Topic created here, along with 16 Releases in different combinations of user roles and approval statuses (as seen the the screenshot attached to the PR).

Once SOW4 is merged into the dev branch and the SOW4 code deployed to the Dev environment, we will run this test manually (pointed towards Dev) and this will create these Themes, Topics, Publications and Releases on Dev.  Then we won't have to run it again (but if we ever do need to, or if we need to prep any other environment for testing against, we can run this manually and it'll generate what we need).

Reply:
(and it's time consuming mostly due to the file uploads and the approval process needed for creating the Releases)

Reply:
Right - you did say you were going to do this and I totally forgot 🤦